server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -210,7 +211,12 @@ public void setValues(PreparedStatement ps) throws SQLException {
             userDetails.put("verified", existingUser.isVerified());
             userDetails.put("user_id", existingUser.getId());
             throw new ScimResourceAlreadyExistsException("Username already in use: " + existingUser.getUserName(), userDetails);
+        } catch(DataIntegrityViolationException e) {
+            logger.debug("DataIntegrityViolationException thrown: ", e);
+                throw new InvalidScimResourceException("ScimUser:" + user + " is Invalid and can not be created. " +
+                        "Please validate the property values.");
```

Comment:
user should be user.toString() for the exception message. Also fix the capitalization for the message.

Reply:
user.toString() is always implicitly called in string concatenation... 
capitalization is fine I believe. Invalid is capital for emphasis.

Reply:
"ScimUser:org.cloudfoundry.identity.uaa.scim.ScimUser@7a04fea7 is Invalid and can not be created. Please validate the property values." is what is being printed out so needs to be fixed.

Reply:
In that case ScimUser needs to have a toString implementation.  Should be added at the earliest.

Reply:
Updated to be:
"ScimUser:<username_submitted> is invalid and can not be created. Please validate the property values for size and type."